File: array/max_min.js
var arr = [1, 2, 3, 4, 5];

var min = 0, max = arr.length - 1, flag = true;
var new_arr = [];
for (let i in arr) {
    if (flag)
        new_arr.push(arr[max --]);
    else
        new_arr.push(arr[min ++]);
    flag = !flag;
}

console.log(arr);
console.log(new_arr);

File: array/max_min.cpp
#include <bits/stdc++.h>

using namespace std;


int main() {
    int arr[] = {1, 2, 3, 4, 5}, n = 5;

    int min = 0, max = n - 1, flag = true, new_arr[n];
    for (int i = 0; i < n; i ++) {
        if (flag) 
            new_arr[i] = arr[max --];
        else 
            new_arr[i] = arr[min ++];
        flag = !flag;
    }

    for (int i = 0; i < n; i ++) 
        cout << arr[i] << " ";
    cout << endl;
    for (int i = 0; i < n; i ++) 
        cout << new_arr[i] << " ";

    return 0;
}